scenario_player/utils/configuration/nodes.py

```diff
@@ -0,0 +1,70 @@
+from typing import List, Optional, Union
+
+import structlog
+
+from scenario_player.exceptions.config import NodeConfigurationError
+from scenario_player.utils.configuration.base import ConfigMapping
+
+log = structlog.get_logger(__name__)
+
+
+class NodesConfig(ConfigMapping):
+    """Thin wrapper around the 'nodes' setting section of a loaded scenario .yaml file.
+
+    Validates the givne config for missing values and mutually exclusive options.
+
+    :param loaded_yaml: The dict loaded from the scenario yaml file.
+    :param scenario_version: Version of the scenario yaml file.
+    """
+
+    def __init__(self, loaded_yaml: dict):
+        super(NodesConfig, self).__init__(loaded_yaml.get("nodes", {}))
+        self.validate()
+
+    @property
+    def raiden_version(self) -> str:
+        return self.dict.get("raiden_version", "LATEST")
+
+    @property
+    def count(self):
+        return self["count"]
+
+    @property
+    def default_options(self) -> dict:
+        return self.dict.get("default_options", {})
+
+    @property
+    def node_options(self) -> dict:
+        return self.get("node_options", {})
+
+    @property
+    def list(self) -> List[str]:
+        """Return the list of nodes configured in the scenario's yaml."""
+        return self["list"]
+
+    @property
+    def commands(self) -> dict:
+        """Return the commands configured for the nodes."""
+        return self.get("commands", {})
+
+    @staticmethod
+    def assert_option(expression, err: Optional[Union[str, Exception]] = None):
+        """Assert the given expression and raise a NodeConfigurationError if it fails."""
+        if isinstance(err, str):
+            err = NodeConfigurationError(err)
+        return ConfigMapping.assert_option(expression, err)
+
+    def validate(self):
```

Comment:
This should also validate that `mode: managed` is specified (unless we want to make this optional since it's the only option for v2 currently).

Reply:
I've intentionally dropped it, since there were only two modes (external, managed) - since `external` was v1 only, it essentially boils down to being just `managed` - which is now implicit, and no longer required.